src/model/custom_electra/electra.py

```diff
@@ -0,0 +1,114 @@
+
+    def __init__(
+        self,
+        generator_model: transformers.DebertaV2Config,
+        discriminator_model: transformers.DebertaV2Config,
+        vocab_size: int,
+        embedding_sharing: bool= True,
+        discriminator_lambda_loss: float= 50.0, 
+        pretrained_model_name_or_path= None,
+        **kwargs,
+    ):  
+        print("===== kwargs =====")
+        for key, value in kwargs.items():
+            print(f"{key}: {value}")
+        print(f'pretraineds_model_name_or_path: {pretrained_model_name_or_path}')
+        #super().__init__(pretrained_model_name_or_path)
+        super().__init__(**kwargs)
+        self.vocab_size= vocab_size
+        self.generator_config= generator_model
+        self.discriminator_config= discriminator_model
+        self.embedding_sharing= embedding_sharing
+        self.discriminator_lambda_loss= discriminator_lambda_loss
+        self._name_or_path= pretrained_model_name_or_path
+        
+        #self._name_or_path= CustomElectraConfig 
+# suggest to subclass PreTrainedModel, this has several advantages to using nn.Module
+# you can take advantage of existing functions and ensure it will work with Trainer
+# https://discuss.huggingface.co/t/resources-for-using-custom-models-with-trainer/4151
+class CustomElectraModel(PreTrainedModel):
+
+    # loss weights are default from paper
+    #def __init__(self, config: AutoConfig, tokenizer: PretrainedTokenizerFast=None, lambda_loss=50.0):
+    
+    config_class = CustomElectraConfig
+
+    def __init__(self, config: CustomElectraConfig):
+
+        super().__init__(config)
+
+        self.lambda_loss = self.config.discriminator_lambda_loss
+        self.gumbel_dist = torch.distributions.gumbel.Gumbel(0.,1.)
+
+        # can change this to pass the configs,
+        # or pass the generator/discriminators themselves
+        #base_gen_config = ElectraConfig.from_pretrained('google/electra-base-generator')
+        base_gen_config= AutoConfig.from_pretrained(config.generator_config)
+        self.generator: ElectraForMaskedLM = AutoModelForMaskedLM.from_config(base_gen_config)
+
+        #base_disc_config = ElectraConfig.from_pretrained('google/electra-base-discriminator')
+        base_disc_config = AutoConfig.from_pretrained(config.discriminator_config)
+        self.discriminator: ElectraForPreTraining = AutoModelForMaskedLM.from_config(base_disc_config)
+
+        # we can modify src/train_tokenizer.py to take different tokenizer classes
+        # and train it there, pass it here
+        #self.tokenizer = tokenizer if tokenizer else PretrainedTokenizerFast.from_pretrained("google/electra-base-generator")
+
+    
+    def forward(self, 
+                input_ids: Optional[torch.Tensor] = None, 
+                attention_mask: Optional[torch.Tensor] = None, 
+                token_type_ids: Optional[torch.Tensor] = None, 
+                labels: Optional[torch.Tensor] = None) -> ElectraForPreTrainingOutput:
```

Comment:
As per above: return type could be `TokenClassifierOutput`: https://huggingface.co/docs/transformers/v4.44.2/en/main_classes/output#transformers.modeling_outputs.TokenClassifierOutput